### ISSUE TITLE
Unify styling of file upload buttons with drop zone

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#65333]
Related to #62743

## What problem does this address?

I noticed we have two "file upload" buttons that have different custom styling for no apparent reason. This is non-ideal for both code maintainability and UX.

### Font upload

https://github.com/user-attachments/assets/f74b8d2a-c4d2-433e-9177-804bcb453b2d

### Set featured image

https://github.com/user-attachments/assets/c24ea654-7092-4812-b9a1-6dd78c5f0461

## What is your proposed solution?

Possible solutions include:

- Use the standard `"secondary"` Button variant.
- Introduce a new Button variant with guidance that it should be used for file upload buttons, and use that consistently.
- Introduce a new UI pattern that denotes a "file upload button with drop zone". For reference, here is another one of these that appears in a block placeholder:

    <img width="652" alt="Featured image block placeholder" src="https://github.com/user-attachments/assets/046891a2-59f3-48c7-acf5-b1635e7dbbed">

cc @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
> Introduce a new Button variant with guidance that it should be used for file upload buttons, and use that consistently.

Could it make sense to update [`DropZone`](https://wordpress.github.io/gutenberg/?path=/docs/components-dropzone--docs) for this purpose, so that it behaved more like [this example](https://polaris.shopify.com/components/selection-and-input/drop-zone?example=drop-zone-default) in Polaris? The option to include help text would be handy, as would the sizing props.

Additionally we might include a state where the uploaded file(s)a are presented in the UI with actions to reset / replace, this could address the a11y issues in https://github.com/WordPress/gutenberg/issues/62743.

**Comment by mirka** [comment]
> Could it make sense to update [`DropZone`](https://wordpress.github.io/gutenberg/?path=/docs/components-dropzone--docs) for this purpose, so that it behaved more like [this example](https://polaris.shopify.com/components/selection-and-input/drop-zone?example=drop-zone-default) in Polaris? The option to include help text would be handy, as would the sizing props.

We might want to start by spec'ing a UI pattern that works for all the instances in the OP, and document that pattern as a story for DropZone.

> Additionally we might include a state where the uploaded file(s)a are presented in the UI with actions to reset / replace, this could address the a11y issues in #62743.

This could also start as a pattern documented in a story somewhere. I don't think we have enough occurrences yet to commit to an API.

**Comment by jameskoster** [comment]
Something like this might work:

<img width="1173" alt="DropZone" src="https://github.com/user-attachments/assets/f5249199-6a36-4b05-9f59-fd69ee385fed">


It's essentially one big `secondary` button with design cues from https://github.com/WordPress/gutenberg/issues/64744.

<img width="1228" alt="insitu" src="https://github.com/user-attachments/assets/2399a63a-a72d-4b21-ae36-261ed6ff596c">

The on-canvas version may need a different treatment to ensure visibility against a multitude of backgrounds.

**Comment by mmjoubert** [comment]
It still seems unclear that a "standard" design offers a drop-to-upload option. I think it would be more helpful to abandon the "standard" design and combine the "With icon button" and "With help text" designs to create a whole new "standard" design.

**Comment by iamleese** [comment]
I agree to standardize the look of the DropZone, however we need to address smaller drop zones like the "Set featured image"

From an accessibility standpoint, it's hard for users with impaired vision and motor skill issues to focus a drop in a small area. Previously, the featured image area used to have a larger box where it was obvious where you could drop an image, at the same time, there was a button below to select images from the media library.

Taking the suggestion of @jameskoster maybe we can revert back to the former style but using a standardized look for the drop zone.

<img width="410" alt="featuredimaged_drop" src="https://github.com/user-attachments/assets/f1d36eda-5f29-4313-8762-cdb0f930fdff">

**Comment by richtabor** [comment]
> Previously, the featured image area used to have a larger box where it was obvious where you could drop an image, at the same time, there was a button below to select images from the media library.
> 
> Taking the suggestion of @jameskoster maybe we can revert back to the former style but using a standardized look for the drop zone.

You can also select this featured image DropZone area to open the Media Library. It's unlikely it needs duplicative UI.

**Comment by afercia** [comment]
See related https://github.com/WordPress/gutenberg/issues/65745 about localization issues.

**Comment by jameskoster** [comment]
Some additional considerations for the design of a new DropZone/Upload button:

- What does the DropZone/Upload button look like when a file has been selected but not uploaded?
- How does the user deselect or remove the uploaded file?
- Does it need a busy state, if so how does that look?
- What does it looks like when a file has been uploaded?
- What about multiple files?
- Should consumers have the ability to provide users with upload options; open media library / upload directly from the computer / both?

**Comment by afercia** [comment]
> From an accessibility standpoint, it's hard for users with impaired vision and motor skill issues to focus a drop in a small area. Previously, the featured image area used to have a larger box 

Totally agree. The redesign fo the Post Summary made the Featured image drop zone ways smaller, which I'd consider an usability and accessibility regression. Any drop zone needs to be bigger.

**Comment by ciampo** [comment]
I'm now trying to take sides, but could we consider drag-and-drop just an alternative to clicking? Just a way to offer different interactions that achieve the same purpose, ie. layers of redundancy. Users with impaired vision and motor skills would be able to interact with the UI widget by clicking the button.

**Comment by jameskoster** [comment]
If we treat the dropzone as a regular button (in terms of visuals) then how would users know they can drop files on it? I have to say, looking at the featured image upload button in the OP I would not interpret it is a drop zone.

A minimum size (overall area) for dropzones seems reasonable, but I can't find any guidelines about that.

**Comment by jameskoster** [comment]
Thinking about this one a little more, I wonder if a new component might make sense. Something like '`FileControl`', which would essentially be a wrapper for the native `input type="file"` with an enhanced design that makes use of `DropZone`, plus any other affordances we deem worthwhile. 

However I also realise that many use cases for such a component in WP would require coupling with the media library and/or post meta. For example consider the control for setting a featured image; once you drop a file on it, that file is simultaneously uploaded to the media library and set as the 'post thumbnail'. This all occurs automatically without having to submit the file. Would this integration with the media library be something that is built into the component, or would it be something that consumers extend the component to add ad hoc?

**Comment by ciampo** [comment]
> Thinking about this one a little more, I wonder if a new component might make sense. Something like 'FileControl', which would essentially be a wrapper for the native input type="file" with an enhanced design that makes use of DropZone, plus any other affordances we deem worthwhile.

That could be a good idea, since it would leverage native platform features, like native drag-n-drop and clicability — although we'd need to look into it a bit more.

> However I also realise that many use cases for such a component in WP would require coupling with the media library and/or post meta. For example consider the control for setting a featured image; once you drop a file on it, that file is simultaneously uploaded to the media library and set as the 'post thumbnail'. This all occurs automatically without having to submit the file. Would this integration with the media library be something that is built into the component, or would it be something that consumers extend the component to add ad hoc?

We'd keep the WP-specific part of of the component. Consumers would be able to listen to events, ie. `onChange` (or any other name), and use the callback data to take actions (ie. upload to media library, set as thumbnail, etc). This is very similar to how `DropZone` already works via the `onDrop` `onFilesDrop` and `onHTMLDrop` props

**Comment by jameskoster** [comment]
> We'd keep the WP-specific part of of the component

Just clarifying; you mean the wp-specific part would be kept _out_ of the component, right? :)
